drivers/soundwire/intel_init.c

```diff
@@ -65,6 +65,8 @@ static int sdw_intel_cleanup(struct sdw_intel_ctx *ctx)
 		if (md) {
 			md->driver->remove(md);
 			put_device(&md->dev);
```

Comment:
@bardliao is the put_device() still necessary if you unregister?

Reply:
in my tests the put_device() is indeed not useful, but the md->dev.driver assignment is required to avoid a kernel oops. I still need to understand that part, it's a bit shady to me what we are doing with manual assignments of the dev.driver field.

Reply:
@plbossart I've been thinking about the dev.driver part too since you posted the patches and my theory is that it due to this: https://github.com/thesofproject/linux/blob/6fc7eb30029afb1fad671a72c7171757f154dfd7/sound/soc/soc-core.c#L1019

If we add a check for component->driver in https://github.com/thesofproject/linux/blob/6fc7eb30029afb1fad671a72c7171757f154dfd7/sound/soc/soc-component.c#L300, we might be able to avoid this. 

But then this would mean that the module would be able to be removed when the component is in use. So Im not sure